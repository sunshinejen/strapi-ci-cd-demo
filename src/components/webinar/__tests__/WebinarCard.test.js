import React from 'react';
import { render, screen, fireEvent, getByTestId } from '@testing-library/react';
import WebinarCard from '../webinarCard';
import { Modal } from 'react-bootstrap';

//webinar test data 
const webinarTestData = {
    date: 'Thursday, July 09, 2037',
    webinar_title: 'Racism, Why We Must Solve It in Technology',
    webinar_description: "We explore how can technology improve the treatment of racial groups and their interactions with the Police, and criminal justice system, improve equity in education, improve financial opportunities and more.",
    flyer_img: {
        url: "https://res.cloudinary.com/dlz977spb/image/upload/v1659651175/racism_solve_in_tech_flyer_bfdfe16a62.png"
    },
    registration: "https://docs.google.com/forms/d/e/1FAIpQLSeV3qKNMJDTZ6jb1sKlesJNCU4p_x4D0r5ZUUZ3VRgHsznJ1g/viewform",
    webinar_url: "https://www.youtube.com/embed/XE-pTIqLTwE",
    panelist_names: "test",
    moderator_names: "names",
    key_takeaways: "testing"
}


//Smoke Test
it("Webinar Card renders as expected", () => {
    render(<WebinarCard webinar={webinarTestData} />)
});

test("Webinar Card snapshot", () => {
    const { asFragment } = render(<WebinarCard webinar={webinarTestData} />)
    expect(asFragment()).toMatchSnapshot()
});

test("Card Unit Test", () => {
    const { getByText,getByTestId } = render(<WebinarCard
        webinar={webinarTestData}
    />)

    expect(getByText("Thursday, July 09, 2037")).toBeInTheDocument()
    expect(getByText('Racism, Why We Must Solve It in Technology')).toBeInTheDocument()
    expect(getByText("We explore how can technology improve the treatment of racial groups and their interactions with the Police, and criminal justice system, improve equity in education, improve financial opportunities and more.")).toBeInTheDocument()

    //img 
    const testImage = document.querySelector("img")
    expect(testImage.src).toContain("https://res.cloudinary.com/dlz977spb/image/upload/v1659651175/racism_solve_in_tech_flyer_bfdfe16a62.png")
   

    //modal open
    fireEvent.click(screen.getByTestId('webinar-view-button'))
    expect(getByText("test")).toBeInTheDocument()
    expect(getByText("names")).toBeInTheDocument()
    expect(getByText("testing")).toBeInTheDocument()
    expect(getByTestId("modal-description")).toBeVisible()
    expect(getByTestId("modal-title")).toBeVisible()
    expect(getByTestId("modal-panelist")).toBeVisible()
    expect(getByTestId("modal-moderators")).toBeVisible()
    expect(getByTestId("modal-takeaways")).toBeVisible()



    //iframe 
    const testIframe = document.querySelector("iframe")
    expect(testIframe.src).toContain("https://www.youtube.com/embed/XE-pTIqLTwE")

    //modal close
    fireEvent.click(screen.getByTestId('webinar-view-button'))
})
