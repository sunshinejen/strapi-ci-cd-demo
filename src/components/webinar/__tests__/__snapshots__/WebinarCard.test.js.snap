// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webinar Card snapshot 1`] = `
<DocumentFragment>
  <div
    class="web-grid"
  >
    <div
      class="webcard"
    >
      <img
        alt="webinar poster"
        class="webCard-img-top"
        src="https://res.cloudinary.com/dlz977spb/image/upload/v1659651175/racism_solve_in_tech_flyer_bfdfe16a62.png"
      />
      <div
        class="webcard-body"
      >
        <h6
          class="webcard-subtitle mb-2 text-muted"
        >
          Thursday, July 09, 2037
        </h6>
        <h5
          class="webcard-title"
        >
          Racism, Why We Must Solve It in Technology
        </h5>
        <div
          class="webcard-text"
        >
          <p>
            We explore how can technology improve the treatment of racial groups and their interactions with the Police, and criminal justice system, improve equity in education, improve financial opportunities and more.
          </p>
        </div>
        <button
          class="readMoreBtn btn-link"
          data-testid="webinar-view-button"
          style="color: rgb(175, 134, 97); text-decoration: none;"
          type="button"
        >
          View Webinar
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
